Cosmos3 Distilled support (#14177)
huggingface/diffusers: docs/source/en/api/pipelines/cosmos3.md

export_to_video(videos, "cosmos3_modular_transfer_edge.mp4", fps=30, macro_block_size=1)
```

### Distilled (few-step) text-to-image and image-to-video

Few-step distilled checkpoints ship a fixed sigma schedule in

> `scheduler.config.fixed_step_sampler_config.t_list` and bake classifier-free guidance into

Reviewer: I think we need to udpate the doc here, no? now it is a pipeline config

the weights. They are served by the dedicated [`Cosmos3DistilledModularPipeline`] (blocks:
`Cosmos3DistilledBlocks`) — the task-based [`Cosmos3OmniPipeline`] and the base
[`Cosmos3OmniModularPipeline`] do not implement the distilled contract.

Load a distilled repo and call the pipeline **without** `num_inference_steps` or
`guidance_scale`; `Cosmos3DistilledSetTimestepsStep` reads the fixed step count from the
scheduler config and forces `guidance_scale=1.0`. Because classifier-free guidance is baked

> into the weights, `negative_prompt` is not supported (passing one raises an error).

Reviewer: ```suggestion
into the weights, `negative_prompt` is not supported (passing one raises a warning and get ignored).
```

pipe.to("cuda")

# text-to-image (distilled)
videos = pipe(
    prompt=json.dumps(json_prompt),
    num_frames=1,
    height=720,
    width=1280,
    output="videos",
)
videos[0].save("cosmos3_distilled_t2i.jpg", format="JPEG", quality=85)

# image-to-video (distilled) — load the I2V repo instead
# JSON-upsampled prompt (see "Prompt upsampling" above); upsampled from the source prompt
# "The right robotic hand picks up the red sphere on the shelf."
json_prompt_i2v = json.load(open("assets/example_i2v_prompt.json"))

repo_i2v = "nvidia/Cosmos3-Super-Image2Video-4Step"
pipe = Cosmos3DistilledModularPipeline.from_pretrained(repo_i2v, torch_dtype=torch.bfloat16)
pipe.load_components(torch_dtype=torch.bfloat16)
pipe.to("cuda")

image = load_image(
    "https://github.com/nvidia-cosmos/cosmos-dependencies/raw/refs/heads/assets/cosmos3/inputs/vision/robot_153.jpg"
)
videos = pipe(
    prompt=json.dumps(json_prompt_i2v),
    image=image,
    num_frames=189,
    height=720,
    width=1280,
    output="videos",
)
export_to_video(videos, "cosmos3_distilled_i2v.mp4", fps=24, macro_block_size=1)
```

For a CLI wrapper with warmup / iteration timing, see

> [`examples/cosmos3/inference_cosmos3_modular_distilled.py`](../../../examples/cosmos3/inference_cosmos3_modular_distilled.py).

Reviewer: This link does not exist
